exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1188 @@
+			orderStatus = order.Cancelled
+		} else {
+			orderStatus, err = order.StringToOrderStatus(strings.ToUpper(response[i].State))
+			if err != nil {
+				log.Errorf(log.ExchangeSys, "%s %v", ok.Name, err)
+			}
+		}
+		if orderStatus == order.New {
+			continue
+		}
+		pair, err := ok.GetPairFromInstrumentID(response[i].InstrumentID)
+		if err != nil {
+			log.Errorf(log.ExchangeSys, "%s %v", ok.Name, err)
+			continue
+		}
+		if !(len(req.Pairs) == 0 || len(req.Pairs) == 1 && req.Pairs[0].Equal(currency.EMPTYPAIR)) {
+			x := 0
+			for x = range req.Pairs {
+				if req.Pairs[x].Equal(pair) {
+					break
+				}
+			}
+			if len(req.Pairs) == x {
+				continue
+			}
+		}
+		oType, err := ok.OrderTypeFromString(response[i].OrderType)
+		if err != nil {
+			return nil, err
+		}
+		orders[i] = order.Detail{
+			Amount:          response[i].Size,
+			Pair:            pair,
+			Price:           response[i].Price,
+			ExecutedAmount:  response[i].FillSize,
+			RemainingAmount: response[i].Size - response[i].FillSize,
+			Fee:             response[i].FeeCurrency,
+			Exchange:        ok.Name,
+			OrderID:         response[i].OrderID,
+			ClientOrderID:   response[i].ClientSupplierOrderID,
+			Type:            oType,
+			Side:            orderSide,
+			Status:          orderStatus,
+			AssetType:       req.AssetType,
+			Date:            response[i].CreationTime,
+			LastUpdated:     response[i].UpdateTime,
+		}
+	}
+	return orders, nil
+}
+
+// GetFeeByType returns an estimate of fee based on the type of transaction
+func (ok *Okx) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	if feeBuilder == nil {
+		return 0, fmt.Errorf("%T %w", feeBuilder, common.ErrNilPointer)
+	}
+	if !ok.AreCredentialsValid(ctx) && feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		feeBuilder.FeeType = exchange.OfflineTradeFee
+	}
+	return ok.GetFee(ctx, feeBuilder)
+}
+
+// ValidateCredentials validates current credentials used for wrapper
+func (ok *Okx) ValidateCredentials(ctx context.Context, assetType asset.Item) error {
+	_, err := ok.UpdateAccountInfo(ctx, assetType)
+	return ok.CheckTransientError(err)
+}
+
+// GetHistoricCandles returns candles between a time period for a set time interval
+func (ok *Okx) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if err := ok.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+	if kline.TotalCandlesPerInterval(start, end, interval) > float64(ok.Features.Enabled.Kline.ResultLimit) {
+		return kline.Item{}, errors.New(kline.ErrRequestExceedsExchangeLimits)
+	}
+	instrumentID, err := ok.getInstrumentIDFromPair(ctx, pair, a)
+	if err != nil {
+		return kline.Item{}, err
+	}
+	candles, err := ok.GetCandlesticksHistory(ctx, instrumentID, interval, time.Time{}, time.Time{}, 0)
```

Comment:
The maximum is 300. Please update `ok.Features.Enabled.Kline.ResultLimit`. The API https://www.okx.com/docs/en/#rest-api-market-data-get-candlesticks is saying that it can retreive the latest 1440, but the limit of the request is 300 of those 1440

```suggestion
	if kline.TotalCandlesPerInterval(start, end, interval) > float64(ok.Features.Enabled.Kline.ResultLimit) {
		return kline.Item{}, errors.New(kline.ErrRequestExceedsExchangeLimits)
	}
	instrumentID, err := ok.getInstrumentIDFromPair(ctx, pair, a)
	if err != nil {
		return kline.Item{}, err
	}
	candles, err := ok.GetCandlesticksHistory(ctx, instrumentID, interval, time.Time{}, time.Time{}, ok.Features.Enabled.Kline.ResultLimit)
```
Using that limit here updates it from 100, to 300